app/src/main/java/ch/epfl/sdp/appart/location/AndroidLocationService.java

```diff
@@ -106,22 +132,39 @@ public void onLocationResult(@NonNull LocationResult locationResult) {
     }
 
     @Override
-    public Location getLocationFromName(String name) {
-        if (name == null) return null;
-
-        final List<Address> addresses;
+    public CompletableFuture<Location> getLocationFromName(String address) {
+        CompletableFuture<Location> futureLocation = new CompletableFuture<>();
         try {
-            addresses = this.geocoder.getFromLocationName(name, 1);
-        } catch (IOException e) {
-            e.printStackTrace();
-            return null;
-        }
-        if (addresses.isEmpty()) return null;
+            String url = "https://maps.googleapis" +
+                    ".com/maps/api/geocode/json?address=" + URLEncoder.encode(address, "UTF-8") + "&key=" + api_key;
```

Comment:
In a refactoring PR we could put these values into constants

Reply:
Yes, indeed.